### PR TITLE
feat: pull gas info field from subgraph

### PR DESCRIPTION
**Pull request by jerrylaloan**
## Description of the changes

ticket: https://app.asana.com/0/1187210138355031/1202055267731943/f

- pull `gasUsed` & `gasPrice` field from updated payment subgraph graphql query
- store fields into request object

### CODE REVIEW

File: packages/payment-detection/src/thegraph/queries/GetPaymentsAndEscrowState.graphql [hunk 1]
```diff
@@ -28,5 +30,7 @@ query GetPaymentsAndEscrowState(
     from
     timestamp
     block
+    gasUsed
+    gasPrice
```
Comment (benjlevesque) [inline packages/payment-detection/src/thegraph/queries/GetPaymentsAndEscrowState.graphql:34]: These fields are not available for escrow. I'm not sure it makes sense to add them 🤔

File: packages/payment-detection/test/erc20/fee-proxy-contract.test.ts [hunk 2]
```diff
@@ -282,4 +282,61 @@ describe('api/erc20/fee-proxy-contract', () => {
...
+      timestamp: 0,
+      version: '0.2',
+    };
+
+    erc20FeeProxyContract = new ERC20FeeProxyPaymentDetector({
+      advancedLogic: mockAdvancedLogic,
+      currencyManager,
+    });
+
+    const balance = await erc20FeeProxyContract.getBalance(mockRequest);
```
Comment (jerrylaloan) [inline packages/payment-detection/test/erc20/fee-proxy-contract.test.ts:334]: should I also mock the `extractEvents` result for this test? @benjlevesque

Comment (benjlevesque) [reply packages/payment-detection/test/erc20/fee-proxy-contract.test.ts:334]: if this test is running without mock, I believe you can yes!

don't try to write the mocked data yourself, copy the exact return value from that function 🙏

File: packages/payment-detection/src/thegraph/queries/GetPaymentsAndEscrowState.graphql (hunk 1 above)
Comment (jerrylaloan) [reply packages/payment-detection/src/thegraph/queries/GetPaymentsAndEscrowState.graphql:34]: actually it's available on escrowEvents. but my question is, does the escrowEvents don't need those fields? if yes, maybe I made a mistake on my change on the subgraph repo...

**Review by benjlevesque — APPROVED** [review]

File: packages/payment-detection/test/erc20/fee-proxy-contract.test.ts [hunk 3]
```diff
@@ -282,4 +282,86 @@ describe('api/erc20/fee-proxy-contract', () => {
...
+                paymentAddress: '0x6d69c636c825263Aa71a3D86D32D0E4897a7a580',
+                salt: '4f43d8b97ae1f63d',
+              },
+              timestamp: 1655883560,
+            },
+          ],
+          id: ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_FEE_PROXY_CONTRACT,
+          type: ExtensionTypes.TYPE.PAYMENT_NETWORK,
+          values: {
+            // refundAddress: '0xrefundAddress',
```
Comment (kevindavee) [inline packages/payment-detection/test/erc20/fee-proxy-contract.test.ts:313]: Remove comment?

**Review by kevindavee — APPROVED** [review]

**Pull request merged**